public/Get-DbaFile.ps1

```diff
@@ -191,7 +191,14 @@ function Get-DbaFile {
                     foreach ($type in $FileTypeComparison) {
                         if ($row.filename.ToLowerInvariant().EndsWith(".$type")) {
                             $fullpath = $row.fullpath.Replace("\", $separator)
+                            
+                            # Replacing all instances of '\\' with single backslashes '\', and maintain the leading SMB share path represented by the initial '\\'.
+                            $is_smb_share_path = $fullpath.SubString(0, 2) -eq "\\"
                             $fullpath = $fullpath.Replace("\\", "\")
+                            if ($is_smb_share_path) {
+                                $fullpath = $fullpath -replace "^\\", "\\"
+                            }
+                            
```

Comment:
I am trying to remember why even try to change the path separator from what SQL returns 🤔

Reply:
In SQL Server, you can create a database like this:

```

CREATE DATABASE [test]
ON PRIMARY
(
    NAME = N'test',
    FILENAME = N'C:\\Program Files\\Microsoft SQL Server\\MSSQL16.MSSQLSERVER\\MSSQL\\DATA\\test.mdf',
    SIZE = 8192KB,
    FILEGROWTH = 65536KB
)
LOG ON
(
    NAME = N'test_log',
    FILENAME = N'C:\\Program Files\\Microsoft SQL Server\\MSSQL16.MSSQLSERVER\\MSSQL\\DATA\\test_log.ldf',
    SIZE = 8192KB,
    FILEGROWTH = 65536KB
);
```

After creating the database, if you query the sys.database_files DMV, you will get a physical_name like `C:\\Program Files\\Microsoft SQL Server\\MSSQL16.MSSQLSERVER\\MSSQL\\DATA\\test.mdf`. Also, if there are flaws in your script when moving database files, you could encounter the same issues. It seems that the replace operation is intended to address these issues.